docs2/site/docs/getting-started/dependency-injection.md

```diff
@@ -110,3 +110,128 @@ public void Install(IWindsorContainer container, IConfigurationStore store)
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    ...
+
+    // Add SteroidsDI Open Generic Defer<> Factory Class
+    services.AddDefer();
+
+    // Add SteroidsDI IScopeProvider to use the AspNetCoreHttpScopeProvider
+    // which internally uses the IHttpContextAccessor.HttpContext.RequestServices;
+    services.AddHttpScope();
+
+    ...
+}
+```
+
+Then in your query graph types you can request services using `Defer<T>` to be injected via DI,
+ which will be evaluated at runtime to their relevant Scoped services, `T` has been registered with a `Scoped` DI lifetime:
+
+```csharp
+public class StarWarsQuery : ObjectGraphType
+{
+  // #1 - Add dependencies using Defer<T>
+  public StarWarsQuery(Defer<IDroidRepo> repoFactory)
+  {
+    Field<DroidType>(
+      "hero",
+
+      // #2 Resolve dependencies using current scope provider
+      resolve: context => repoFactory.Value.GetDroid("R2-D2")
+
+    );
+  }
+}
+```
+
+1. Add `Defer<T>` to be injected by the dependency injection container. This is a factory which upon calling `Defer.Value` will resolve the requested service using any currently registered scope provider (e.g. `AspNetCoreHttpScopeProvider`)
+2. Use the `Defer<T>` factory class to resolve the requested dependency using any currently registered scope provider. In our case it will attempt to use the `IHttpContextAccessor.HttpContext.RequestServices` which is the ASP.NET Core Scoped `IServiceProvider` in order to resolve the dependency.
+
+## UserContext Scoped IServiceProvider
+
+**Be Aware**: Exposing the service locator is considered an Anti-Pattern. See [Service Locator is an Anti-Pattern](https://blog.ploeh.dk/2010/02/03/ServiceLocatorisanAnti-Pattern/)
+
+You can add the relevant scoped `IServiceProvider` when creating the `UserContext` in order to run `IDocumentExecuter`.
+
+E.g. In ASP.NET Core [GraphQl.Harness](https://github.com/graphql-dotnet/graphql-dotnet/tree/master/src/GraphQL.Harness) example, you could add `IServiceProvider` to the [GraphQLUserContext](https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL.Harness/GraphQLUserContext.cs) class, and then set it to the `HttpContext.RequestServices` when configuring the `GraphQLSettings.BuildUserContext`
+
+
+1. Add ServiceProvider to the UserContext
+
+    ```csharp
+    public class GraphQLUserContext: Dictionary<string, object>
+    {
+        ...
+        IServiceProvider ServiceProvider { get; set; }
+        ...
+    }
+    ```
+
+2. Set it the the HttpContext.RequestServices
+
+    ```csharp
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.Configure<GraphQLSettings>(settings => settings.BuildUserContext = ctx => new GraphQLUserContext
+        {
+            User = ctx.User,
+            ServiceProvider = ctx.RequestServices
+        });
```

Comment:
Note that the `BuildUserContext` property only exists within the `Harness` sample project, as the entire `GraphQLSettings` class is simply part of the sample.  This may be confusing to people that did not build their code from the sample (but it's pretty easy to discover that).

Reply:
Yeah I noticed that, but I wanted to stay with the samples in this repo, and wasn't sure how to specify the intent without needing to provide a massive code sample. Even the `GraphQLMiddleware` as far as I am aware is only in the harness (I copied mine from there)

Reply:
I will add a note, specifying this.

Reply:
Added, I have split the extension method out, to make it clear to anyone using it, which interfaces/services it uses.